### PR TITLE
Bidirectional Transformer

### PR DESCRIPTION
**Pull request by jrevansjr**
I am proposing to add the following files:
1. Mask Generator that randomly masks j_i based on hyperparameters that can be adjusted for more or less masking.
2. BidirectionalTransformer - Vanilla Transformer
3. Separate pretraining process that could be leveraged with ST as well.

### CODE REVIEW

File: utils/maskgenerator.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            for t in range(T):
+                if torch.rand(1).item() < self.time_step_mask_prob:
+                    joints_to_mask = torch.rand(self.num_joints) < self.joint_mask_prob
+                    indices_to_mask = torch.where(joints_to_mask)[0]
+                    for n in indices_to_mask:
+                        start_idx = n * self.joint_dim
+                        end_idx = start_idx + self.joint_dim
+                        masked_batch_sequence[b, t, start_idx:end_idx] = self.mask_value
+
+        return masked_batch_sequence
```
Comment (moduIo) [inline utils/maskgenerator.py:42]: Could we rename this file `mask_generator.py`? And add a \n (new line) wherever there's a red minus sign in your code?

File: pretrain.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import torch
+
+from models.BiDirectionalTransformer import BiDirectionalTransformer
+from utils.maskgenerator import BatchJointMaskGenerator
+from utils.dataset import prepare_dataset
+from utils.types import ModelEnum, TargetEnum
+
+
+def main(model_name: str = None, target_type: str = None) -> None:
```
Comment (moduIo) [inline pretrain.py:9]: Is there anyway we can merge pretrain.py into the current main.py? It seems pretty similar to me

**Review by moduIo — CHANGES_REQUESTED** [review]

File: pretrain.py (hunk 2 above)
Comment (jrevansjr) [reply pretrain.py:9]: Will change
